### DISCUSSION TITLE
Prompt("%!I") shows an error / a warning when cancelling

### DISCUSSION THREAD

**Discussion by tozGitHub** [WinEdt-Team/WinEdt-Support#16] [1.2 Bugs and Issues]
Dear Alex, dear community,

Sometimes, I execute  
```
Prompt("%I");
```
and then push the Cancel-button. Then I get an error / a warning. I would not expect an error here, since it seems to me that I do not do anything forbidden. However, I am also not sure what should happen instead. Maybe nothing, but IfOK would return false? Most of the times when I cancel, then I would like the macro to stop. One could also think of assigning the empty string to %I.
![grafik](https://github.com/WinEdt-Team/WinEdt-Support/assets/52357589/ca1ccb38-7ad8-4712-bc78-78f1e80361a2)

WinEdt version: Build: 20230519  (v. 11.1) - 64-bit
Windows version: 10.0.22631 Build 22631.

Best,
Thomas

**Comment by Alex-Simonic** [comment]
I could live with this being as it is. I don't see it as a bug or an issue. In this scenario, `IfOK` cannot be used to stop the execution since the error occurs while parsing string parameters inside a macro call. In the next version, macro execution will halt (as it does now) without any warning message. This is consistent with canceling a prompt inside a macro command. Case closed.
